ngraph/core/include/ngraph/op/adaptive_avg_pool.hpp

```diff
@@ -37,6 +37,10 @@ namespace ngraph
 
                 std::shared_ptr<Node>
                     clone_with_new_inputs(const OutputVector& new_args) const override;
+
+                bool evaluate(const HostTensorVector& outputs,
```

Comment:
Why do we need to have this evaluate method?
Do we have an AdaptiveAvgPool on constant folded branch?

Reply:
Currently AdaptiveAvgPool has no CPU plugin implementation, we need to support FastSCNN from Baidu PaddlePaddle which contains AdaptiveAvgPool op, so we need evaluate method to fallback to ngraph reference implementation.

Reply:
@dmitry-gorokhov Do we have a plan to have native support of this op in the CPU plugin?

Reply:
@ilyachur For sure: https://github.com/openvinotoolkit/openvino/pull/6594

Reply:
If we have a plan to merge CPU implementation soon, I propose to don't add evaluate methods.
@dmitry-gorokhov Do you agree with me?

Reply:
For me it sounds reasonable. I would suggest to wait CPU impl.

Reply:
@dmitry-gorokhov @ilyachur  The reason why we need this op fallback logic to unblock the  adaptive pool2d op conversion and paddlepaddle's ppyolo model conversion and validation. Either cpu plugin or fallback path will work. If team agree on the cpu plugin path, it's also fine to me. And please help review cpu plugin PR ( https://github.com/openvinotoolkit/openvino/pull/6594 ) to unblock. Thanks!